l10n_it_ricevute_bancarie/account/account.py

```diff
@@ -85,3 +86,29 @@ class account_invoice(osv.osv):
     _columns = {
         'unsolved_move_line_ids': fields.many2many('account.move.line', 'invoice_unsolved_line_rel', 'invoice_id', 'line_id', 'Unsolved journal items'),
         }
+
+    def action_cancel(self, cr, uid, ids, context=None):
+        if ids:
+            invoice = self.browse(cr, uid, ids)[0]
+            # we get move_lines with date_maturity and check if they are
+            # present in some riba_distinta_line
+            move_line_model = self.pool['account.move.line']
+            rdml_model = self.pool['riba.distinta.move.line']
+            move_line_ids = move_line_model.search(
+                cr, uid, [('move_id', '=', invoice.move_id.id),
+                          ('date_maturity', '!=', False)])
```

Comment:
On 30 July 2015 at 13:54, Andrea Cometa <notifications@github.com> wrote:

> is there some invoice with no date_maturity on account_move?
>
>
>
When you validate an validate an invoice without payment term, you have
receivable move lines without date_maturity.
But, in case of riba, you always have a payment term, then you always have
date_maturity. So this is ok